test/lightline-gitdiff.vader

```diff
@@ -1,2 +1,22 @@
 Include (Algorithms): algorithm/parse_indicator_group.vader
 Include (Utils): utils/group_at.vader
+
+Execute(write_calculation_to_cache(): given no show_empty_indicators variable):
+  call g:lightline#gitdiff#write_calculation_to_cache(1, 0)
```

Comment:
Thanks a lot for providing the tests. I haven't looked at the code for a while and had to find out the syntax for Vader.vim and understand structure of the test files I wrote originally myself.

To write these tests in a clean manner, we should mock the used algorithm. At the current state, calculation is done on buffer 1 -- to be honest, I don't really know what checked there. The following is a sketch for mocking the algorithm and providing a test for mixed results (such as `A: 1 M: 0 D: 1`); we would need to apply this to the subsequent tests as well:

```diff
diff --git a/test/lightline-gitdiff.vader b/test/lightline-gitdiff.vader
index fc926ff..48c60cf 100644
--- a/test/lightline-gitdiff.vader
+++ b/test/lightline-gitdiff.vader
@@ -1,12 +1,27 @@
 Include (Algorithms): algorithm/parse_indicator_group.vader
 Include (Utils): utils/group_at.vader
 
-Execute(write_calculation_to_cache(): given no show_empty_indicators variable):
+Before :
+  if exists('g:LightlineAlgorithm')
+    unlet g:LightlineAlgorithm
+  endif
+
+Execute(write_calculation_to_cache(): given no show_empty_indicators variable with empty result):
+  let g:LightlineAlgorithm = { -> { 'foo': 0, 'bar': 0 } }
+
   call g:lightline#gitdiff#write_calculation_to_cache(1, 0)
   let actual = get(g:, 'lightline#gitdiff#cache')[1]
 Then (should return no empty indicators):
   AssertEqual {}, actual
 
+Execute(write_calculation_to_cache(): given no show_empty_indicators variable with mixed result):
+  let g:LightlineAlgorithm = { -> { 'foo': 1, 'bar': 0 } }
+
+  call g:lightline#gitdiff#write_calculation_to_cache(1, 0)
+  let actual = get(g:, 'lightline#gitdiff#cache')[1]
+Then (should remove the empty key only):
+  AssertEqual { 'foo': 1 }, actual
+
 Execute(write_calculation_to_cache(): given g:lightline#gitdiff#show_empty_indicators == 0):
   let g:lightline#gitdiff#show_empty_indicators = 0
   call lightline#gitdiff#write_calculation_to_cache(2, 0)
```

Note the new variable `g:LightlineAlgorithm`. While writing this, I figured there is a bug in the plugin. On the contrary to what's written in den readme, it's not possible to provide a custom algorithm because, in vimscript, global functions must start with a capital letter. Because of this, I also suggest the following changes:

```diff
diff --git a/README.md b/README.md
index ab17561..edf0511 100644
--- a/README.md
+++ b/README.md
@@ -106,7 +106,7 @@ By default, the latter one is used because it allows to display modified lines.
 This resembles the default:
 
 ```vim
-let g:lightline#gitdiff#algorithm =
+let g:LightlineAlgorithm =
       \ { buffer -> lightline#gitdiff#algorithms#word_diff_porcelain#calculate(buffer) }
 
diff --git a/autoload/lightline/gitdiff.vim b/autoload/lightline/gitdiff.vim
index c1ec1c3..c0d0b3b 100644
--- a/autoload/lightline/gitdiff.vim
+++ b/autoload/lightline/gitdiff.vim
@@ -37,7 +37,7 @@ function! lightline#gitdiff#write_calculation_to_cache(buffer, soft) abort
     return
   endif
 
-  let l:indicator_values = get(g:, 'lightline#gitdiff#algorithm',
+  let l:indicator_values = get(g:, 'LightlineAlgorithm',
       \ { buffer -> lightline#gitdiff#algorithms#word_diff_porcelain#calculate(buffer) })(a:buffer)
 
   " If the user doesn't want to show empty indicators,
@@ -73,7 +73,7 @@ endfunction
 "
 " In fact, an arbitrary number of changes can be supported. This depends on
 " the algorithm that is used for calculation
-" (`g:lightline#gitdiff#algorithm`). However, this function takes only these
+" (`g:LightlineAlgorithm`). However, this function takes only these
 " types of changes into account b/c it only provides default indicators for
 " these types. If an algorithm does not support a particular type, this is not
 " an issue; if it supports more types than this function, the additional types
```